clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/security/GoogleNamedAccountCredentials.java

```diff
@@ -57,8 +57,16 @@ public GoogleNamedAccountCredentials(String accountName,
       this.requiredGroupMembership = requiredGroupMembership == null ? Collections.emptyList() : Collections.unmodifiableList(requiredGroupMembership);
       this.applicationName = applicationName;
       this.credentials = (projectName != null) ? buildCredentials() : null;
+      this.compute = credentials != null ? credentials.getCompute() : null;
       // TODO(ttomsu): Add support for specifying specific regions, to reduce API calls and making debugging easier.
-      this.regionToZonesMap = (credentials != null && credentials.getCompute() != null) ? queryRegions(credentials.getCompute(), projectName) : Collections.emptyMap();
+      this.regionToZonesMap = Collections.emptyMap();
+    }
+
+    public GoogleNamedAccountCredentials initialize() {
```

Comment:
Why is this better than just doing the same initialization on the final line of the constructor?